cli/azd/cmd/middleware/environment.go

```diff
@@ -0,0 +1,82 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/lazy"
+)
+
+// EnvironmentMiddleware is a middleware that loads the environment when not readily available
+type EnvironmentMiddleware struct {
+	lazyAzdContext *lazy.Lazy[*azdcontext.AzdContext]
+	lazyEnvManager *lazy.Lazy[environment.Manager]
+	lazyEnv        *lazy.Lazy[*environment.Environment]
+	envFlags       internal.EnvFlag
+}
+
+// NewEnvironmentMiddleware creates a new instance of the EnvironmentMiddleware
+func NewEnvironmentMiddleware(
+	lazyAzdContext *lazy.Lazy[*azdcontext.AzdContext],
+	lazyEnvManager *lazy.Lazy[environment.Manager],
+	lazyEnv *lazy.Lazy[*environment.Environment],
+	envFlags internal.EnvFlag,
+) Middleware {
+	return &EnvironmentMiddleware{
+		lazyAzdContext: lazyAzdContext,
+		lazyEnvManager: lazyEnvManager,
+		lazyEnv:        lazyEnv,
+		envFlags:       envFlags,
+	}
+}
+
+// Run runs the EnvironmentMiddleware to load the environment when not readily available
+func (m *EnvironmentMiddleware) Run(ctx context.Context, next NextFn) (*actions.ActionResult, error) {
+	// We already have an environment, skip loading
+	// This will typically be the case when an environment has been created from a previous command like `azd init`
+	env, err := m.lazyEnv.GetValue()
+	if err == nil && env != nil {
```

Comment:
Just wondering - in what case (if any) can `m.lazyEnv.GetValue()` return a nil error but a nil Environment as well?

Reply:
I don't think there is a valid case for this - just making extra sure :)
It should either return a nil environment with an error or an environment without an error.